sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -222,6 +223,13 @@ case class ShowFunctionsCommand(
           case (f, "USER") if showUserFunctions => f.unquotedString
           case (f, "SYSTEM") if showSystemFunctions => f.unquotedString
         }
-    functionNames.sorted.map(Row(_))
+    if (showSystemFunctions) {
+      (functionNames ++
+        StringUtils.filterPattern(Seq("!=", "<>", "between", "case"), pattern.getOrElse("*")))
```

Comment:
shall we add comment like https://github.com/apache/spark/blob/9b8d63e7ce76740eb0fc08fcd2b24849ff7693be/sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala#L117